drivers/dma/dma_stm32_bdma.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2023 Jeroen van Dooren, Nobleo Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef BDMA_STM32_H_
+#define BDMA_STM32_H_
+
+#include <soc.h>
+#include <stm32_ll_bdma.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/dma/dma_stm32.h>
+
+/* Maximum data sent in single transfer (Bytes) */
+#define BDMA_STM32_MAX_DATA_ITEMS	0xffff
+
+struct bdma_stm32_channel {
+	uint32_t direction;
+#ifdef CONFIG_DMAMUX_STM32
+	int mux_channel; /* stores the dmamux channel */
+#endif /* CONFIG_DMAMUX_STM32 */
+	bool source_periph;
+	bool hal_override;
+	volatile bool busy;
+	uint32_t src_size;
+	uint32_t dst_size;
+	void *user_data; /* holds the client data */
+	dma_callback_t bdma_callback;
+};
+
+struct bdma_stm32_data {
+	struct dma_context dma_ctx;
+};
+
+struct bdma_stm32_config {
+	struct stm32_pclken pclken;
+	void (*config_irq)(const struct device *dev);
+	bool support_m2m;
+	uint32_t base;
+	uint32_t max_channels;
+#ifdef CONFIG_DMAMUX_STM32
+	uint8_t offset; /* position in the list of bdmamux channel list */
+#endif
+	struct bdma_stm32_channel *channels;
+};
+
+uint32_t bdma_stm32_id_to_channel(uint32_t id);
+#if !defined(CONFIG_DMAMUX_STM32)
+uint32_t bdma_stm32_slot_to_channel(uint32_t id);
+#endif
+
+typedef void (*bdma_stm32_clear_flag_func)(BDMA_TypeDef *DMAx);
+typedef uint32_t (*bdma_stm32_check_flag_func)(BDMA_TypeDef *DMAx);
+
+
+#ifdef CONFIG_DMA_STM32_V1
```

Comment:
Could a DMA of type DBMA also be of type V1 ?

Reply:
Oops, accidentally erased some of my work. This was already deleted. Fixed and removed..